pattern-library/html/boilerplate-header+footer+content+nav.html

```diff
@@ -0,0 +1,113 @@
+<!--
+edX Pattern Library: Boilerplate HTML Template: Header + Footer
+
+NOTES:
+
+* base template with the additon of how to handle an app-centric header/footer semantically.
+* base template with the additon of how to handle different types of navigation (user vs. app-centric).
+* base template with addition of how to start organizing main vs. supplementary/aside content
+* this assumes that the main and aside content are next to each other in the document's flow. This does not neccessarily have to be the case, but the same markup could be used in more complex cases.
+-->
+
+<!DOCTYPE html>
+<html class="no-js" lang="en">
+    <head>
+        <meta charset="utf-8">
+        <!-- always force latest IE rendering engine or request chrome rrame -->
+        <meta content="IE=edge,chrome=1" http-equiv="X-UA-Compatible">
+        <meta name="viewport" content="width=device-width,initial-scale=1">
+
+        <base href="{base path}">
+
+        <!-- place favicon.ico in the root directory -->
+        <link rel="shortcut icon" href="{favicon path}">
+        <!-- place touch icon in the root directory -->
+        <link rel="apple-touch-icon" href="{icon path}">
+
+        <!-- view/page metadata -->
+        <title>{page/view's title}</title>
+        <meta name="description" content="">
+
+        <!-- place primary assets (needed for or to support base view/page rendering) here -->
+        <link rel="stylesheet" href="{main css path}">
+        <script src="{main js path}"></script>
+    </head>
+
+    <body class="{app/site-section-name} {view-name} {stateful-class}">
+        <!-- CASE: if needed, provide a message for older versions of IE -->
+        <!--[if lt IE 9]>
+            <p class="alert alert-browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> to improve your experience.</p>
+        <![endif]-->
+
+        <div class="nav-skip sr-only">
+            <a class="action-skip" href="#nav-app">Skip to the main navigation</a>
+            <a class="action-skip" href="#content-main">Skip to the main content</a>
+            <a class="action-skip" href="#content-aside">Skip to the aside content</a>
+        </div>
+
+        <!-- wrapper for all UI rendered within the page/view -->
+        <div class="wrapper-view">
+
+            <!-- NOTE: wrapper elements are only needed when they need to manage the grid or larger view/page layout aspects (e.g. spanning the whole viewport) -->
+            <div class="wrapper-header-app">
+                <header class="header-app" role="banner">
```

Comment:
a role of banner is not necessary here.  it is implied by the fact that this is the first header on a page and it is not the child of a section or article.  I would like to see an aria-label on here though: ```aria-label="Site"``` This will be presented to the screen reader user as a "Page banner" landmark (as in the banner for the entire page, that contains page level controls.)  Other acceptable values would be Main, Top or Site.  It often depends on context.  Best to choose something that works in all possible contexts, which is why I think Page is the best choice.